### ISSUE TITLE
Make a dialogue window "The following files are newer on disk" easy to understand

### ISSUE THREAD

**Issue by wachtel1** [godotengine/godot-proposals#3456]
### Describe the project you are working on

2D top-down movement-based bullet-hell with puzzles. I use VS Code in my workflow for multiple monitors setup and I often end up editing the scene files directly in VS Code which leads to the "The following files are newer on disk" dialogue window.

### Describe the problem or limitation you are having in your project

I spent a good couple of minutes trying to remember what options in the "The following files are newer on disk" dialogue window mean:

![files_on_disk](https://user-images.githubusercontent.com/89310925/138243721-3f1c19de-1875-4d9e-aa62-11c8314273b2.png)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Rename the buttons:
1. Resave > Keep the file from the editor
2. Reload > Keep the new file
3. Cancel > is this option needed? There are already two different versions of the file at the same time, the user should decide what file to use.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Change the text on the buttons and remove the Cancel button from this window.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I don't see a workaround except enabling the auto-reload for all files with some plugin, which will make this dialogue window disappear. But it will not make this window easier to understand for new users.

### Is there a reason why this should be core and not an add-on in the asset library?

User-friendliness for new users.

**Comment by Calinou** [comment]
> Resave > Keep the file from the editor
> Reload > Keep the new file

Isn't it the other way around? As I understand it, **Resave** keeps your own changes and discards the ones made externally. **Reload** discards your own changes and keeps the ones made externally.

**Comment by was-games** [comment]
> > Resave > Keep the file from the editor
> 
> > Reload > Keep the new file
> 
> 
> 
> Isn't it the other way around? As I understand it, **Resave** keeps your own changes and discards the changes made externally. **Reload** discards your own changes and keeps the changes made externally.

I tested it before creating the issue, and already completely forgot what meant what. I'll comment again later after testing again.

> As for **Cancel**, we can rename that button to **Do Nothing** to be more explicit.

Yes, great idea!

**Comment by nathanfranke** [comment]
Wouldn't "Do Nothing" just mean "Resave" once you Ctrl+S?

**Comment by MarioLiebisch** [comment]
I'd reword the whole dialog, e.g. due to the caption being pointless (it's a modal dialog with mutliple buttons, so you obviously have to confirm something).

**Title:**
```
File changes detected
```

**Text:**
```
The following files currently opened have been changed by an external program.
Do you want to reload the files to keep these changes or discard them?
```

**Buttons (changed order, `Reload` pre-selected):** 
```
Reload
Discard
Cancel
```
`Cancel` could indeed be called `Do nothing`, but I think `Cancel` would be more common or something a user is used to.

**Comment by Calinou** [comment]
I'm not sure if the words "Reload" and "Discard" are sufficient to describe what will be done either.

**Comment by MarioLiebisch** [comment]
Maybe just use a bit more space above and explain this?

**Comment by was-games** [comment]
Maybe something like this?

Title:
`File changes detected`

Text:
```
The following files currently opened in the editor have been changed by an external program.
Do you want to reload the files to keep these changes or discard them?
```

Buttons (changed order, Reload pre-selected):

```
Reload
Discard the changes
Do nothing
```

There is a lot of space between the buttons, so I think that `Discard the changes` should fit right in and it should be clear enough.
I am all in for `Do nothing` as a third option because it is not clear at all what would `Cancel` do in this case. There are 2 versions of the same file(s) exist and the user is asked to decide what to do with them. And it is not obvious that there is an option to do nothing now and save manually later. `Cancel`, while the people are used to this option, is not clear enough.

For future reference, I will leave the description of the current behavior of the options with screenshots:
1. The parameters of a loaded scene in the editor. I will change the "One Shot" parameter in the VSCode:

**Comment by Calinou** [comment]
> BTW, I can and want to do the pull request myself for whatever changes the community will decide on, which would be my first one :). But I think I'll be able to navigate myself.

@was-games Are you still looking to open a pull request?

Personally, I'd go for the following:

```
Resave → Save Local Changes
Reload → Reload From Disk, Discard Local
Cancel → Do Nothing
```

Note that we use Title Case for buttons in the editor, not Sentence case.

**Comment by was-games** [comment]
@Calinou Definitely yes. I'll go with your naming, Godot-centered, not remote-centered naming seems easiest to understand.

**Comment by Karriert** [comment]
Years later and I still have to check on the internet what the three options are supposed to do.

**Comment by tetrapod00** [comment]
This was changed recently twice: https://github.com/godotengine/godot/pull/90660, https://github.com/godotengine/godot/pull/97461. I think at this point we can say that improvements have been made. While we may end up changing the phrasing again (as commented by Calinou in the latest PR), I don't think it makes sense to keep this issue around indefinitely to track that possibility.
